src/cpu/aarch64/brgemm/brgemm_containers.hpp

```diff
@@ -0,0 +1,129 @@
+
+struct brgemm_desc_container_t {
+public:
+    brgemm_desc_container_t() {}
+    brgemm_desc_container_t(size_t ns) { resize(ns); }
+    void resize(size_t ns) { refs_.resize(ns); }
+    inline const brgemm_t *operator[](int idx) const { return refs_[idx]; }
+
+    bool insert(int idx, brgemm_t &brg) {
+        std::vector<char> dummy_bd_mask;
+        std::vector<brgemm_batch_element_t> dummy_static_offsets;
+        return insert(idx, brg, dummy_bd_mask, dummy_static_offsets);
+    }
+
+    bool insert(int idx, brgemm_t &brg, const std::vector<char> &bd_mask,
+            const std::vector<brgemm_batch_element_t> &static_offsets);
+
+private:
+    std::vector<const brgemm_t *> refs_;
+
+    std::set<brgemm_t> set_;
+    std::vector<std::vector<char>> bd_mask_list_;
+    std::vector<std::vector<brgemm_batch_element_t>> static_offsets_list_;
+};
+
+#define BRGEMM_KERNEL_GLOBAL_STORAGE
+
+struct brgemm_kernel_container_t {
+    brgemm_kernel_container_t() {}
+    brgemm_kernel_container_t(size_t ns) { resize(ns); }
+    void resize(size_t ns) { refs_.resize(ns); }
+    inline const brgemm_kernel_t *operator[](int idx) const {
+        return refs_[idx];
+    }
+
+    status_t insert(int idx, const brgemm_t *brg);
+    static bool brgemm_kernel_cmp(const std::shared_ptr<brgemm_kernel_t> &lhs,
+            const std::shared_ptr<brgemm_kernel_t> &rhs);
+
+private:
+    std::vector<const brgemm_kernel_t *> refs_;
+#ifdef BRGEMM_KERNEL_GLOBAL_STORAGE
+    static std::set<std::shared_ptr<brgemm_kernel_t>,
+            decltype(brgemm_kernel_container_t::brgemm_kernel_cmp) *>
+            set_;
+
+    static utils::rw_mutex_t &rw_mutex() {
+        static utils::rw_mutex_t mutex;
+        return mutex;
+    }
+
+    void lock_write() { rw_mutex().lock_write(); }
+    void unlock_write() { rw_mutex().unlock_write(); }
+
+#else
+    std::set<std::shared_ptr<brgemm_kernel_t>,
+            decltype(brgemm_kernel_container_t::brgemm_kernel_cmp) *>
+            set_;
+    void lock_write() {}
+    void unlock_write() {}
+#endif
+    std::map<const brgemm_t *, const brgemm_kernel_t *> brgemm_map_;
+};
+
+struct brgemm_palette_container_t {
+    typedef std::array<char, 64> S_t;
+
+    brgemm_palette_container_t() {}
+    brgemm_palette_container_t(size_t ns) { resize(ns); }
+    void resize(size_t ns) { refs_.resize(ns); }
+
+    inline const char *operator[](int idx) const { return refs_[idx]->data(); }
+
+    bool insert(int idx, const brgemm_t *brg);
+    bool insert(int idx, const brgemm_t &brg) { return insert(idx, &brg); }
+
+    inline void maybe_tile_configure(int &idx, int new_idx) const {
+        if (idx == new_idx) return;
+        idx = new_idx;
+    }
+
+private:
+    std::vector<const S_t *> refs_;
+    std::set<S_t> set_;
+};
+
+} // namespace brgemm_containers
+
+} // namespace aarch64
+} // namespace cpu
+} // namespace impl
+} // namespace dnnl
+
+#endif // CPU_X64_BRGEMM_BRGEMM_CONTAINERS_HPP
```

Comment:
```suggestion
#endif // CPU_AARCH64_BRGEMM_BRGEMM_CONTAINERS_HPP
```

@vineelabhinav Could you please run clang-format on the files introduced in this PR? I would expect clang-format to catch such issues.

Reply:
@igorsafo Sure.